op-challenger/game/fault/trace/outputs/output_alphabet.go

```diff
@@ -30,6 +31,10 @@ func NewOutputAlphabetTraceAccessor(
 		return nil, err
 	}
 
+	if err := v(ctx, outputProvider); err != nil {
+		return nil, err
+	}
```

Comment:
That requires having the trace provider in the registration code which would hoist it up out of this constructor. At that point, it doesn't make sense to me why we split those constructors out into separate files if most of the construction logic happens at the registry level. I'm open to either way, but let's not waste time bikeshedding over this tiny detail if either work correctly.

Reply:
I don't think getting the separation of concerns right is bike shedding - it's an important part of the design and impacts the maintainability of code.

You're right that the provider isn't available in the register code and also that the register code is doing too much. It also feels really weird to need to a validate function into `NewOutputAlphabetTraceAccessor` that it just calls - we're breaking the encapsulation the `TraceAccessor` provides by exposing the underlying provider and also seems wrong to be running a validation process inside a constructor.

This and the fact that we can't get at the absolute pre-state for the bottom half of the game because we don't have its trace accessor yet is really pointing us to an underlying problem that we really want to be able to access the absolute prestate hash without having the trace provider.  Maybe we need to split out a `AbsolutePrestateProvider` that can be constructed with much less context, and the trace provider implementations could build on top of it?

Even with that, I'm not entirely sure where the validation step should live - it feels wrong to do it in the create step.  Maybe it should be a new method on `GamePlayer` and the coordinator becomes responsible for calling it?